### PR TITLE
Run spellcheck only on changed files in prepare_commit.sh

### PR DESCRIPTION
**Pull request by strk**
Should make a few developers with large number of worktrees happier after merge of GH-54847
\cc @troopa81 @3nids

### CODE REVIEW

**Comment by strk** [comment]
This is not ready yet, I've to only pass shell scripts, not all modified files

**Comment by strk** [comment]
ok now only modified files are sent to shell checker.

File: tests/code_layout/test_shellcheck.sh [hunk 1]
```diff
@@ -1,12 +1,20 @@
 #!/usr/bin/env bash
 
-# This runs shellcheck on all sh files
+# This runs shellcheck given or found sh files
```
Comment (troopa81) [inline tests/code_layout/test_shellcheck.sh:3]:
```suggestion
# This runs shellcheck on arguments files, or all files if none was given
```

**Review by troopa81 — APPROVED** [review]
Just a minor comment. Thanks for being so reactive :+1:

**Comment by strk** [comment]
> Just a minor comment. Thanks for being so reactive 👍

Thank you for approving PRs even while suggesting changes, so we don't have to wait too much to merge :)

File: tests/code_layout/test_shellcheck.sh (hunk 1 above)
Comment (strk) [reply tests/code_layout/test_shellcheck.sh:3]: It's not "all files", is just ".sh" files if you don't pass a list.

**Comment by nyalldawson** [comment]
@strk something here has broken the sipify run with prepare_commit. It used to auto run sipify on the changed files, now it no longer does

**Comment by strk** [comment]
Yes sorry, I'm fixing now

**Pull request merged**